lab_1/main.py

```diff
@@ -1,45 +1,44 @@
-def calculate_frequencies(info):
-    words = info.split(' ')
+
+def calculate_frequences(info):
+    wrongs = "1234567890~@#$%^&*\"[]{}\'/\n:;!?().,<>"
     frequencies_dictionary = {}
-    for count, k in enumerate(words):
-        if words[count] in frequencies_dictionary:
-            frequencies_dictionary[words[count]] += 1
-        else:
-            frequencies_dictionary[words[count]] = 1
+    if type(info) is str:
+        text = info.lower()
+        for i in wrongs:
+            text = text.replace(i, "")
+        words = text.split()
+        for count, k in enumerate(words):
+            if words[count] in frequencies_dictionary:
+                frequencies_dictionary[words[count]] += 1
+            else:
+                frequencies_dictionary[words[count]] = 1
     return frequencies_dictionary
 
 
-pass
+def filter_stop_words(freq_dict, stopwords):
+    frequencies = {}
+    if not stopwords:
+        stopwords = []
+    if freq_dict:
+        frequencies = {key: freq_dict[key] for key in freq_dict if (key not in stopwords) and (type(key) is str)}
+    return frequencies
 
 
-def filter_stop_words(frequencies_dict, stopwords):
-    frequencies = {key: frequencies_dict[key] for key in frequencies_dict if key not in stopwords}
-    sorted_words = sorted(frequencies, key=frequencies.get)
-    sorted_words.reverse()
-    return sorted_words
+def get_top_n(sorted_words, top_n):
+    words = sorted(sorted_words.items(), key=lambda item: item[1], reverse=True)
+    if top_n > len(words):
+        top_n = len(words)
+    output = tuple(words[i][0] for i in range(top_n))
+    return output
 
 
-pass
+def read_from_file(path_to_file, lines_limit):
+    text = open(path_to_file)
+    info = text.read(lines_limit)
+    return info
 
 
-def get_top_n(sorted_words, top_n):
-    out = ''
-    for i in range(top_n):
-        out += sorted_words[i]
-        out += '\n'
-    return out
-
-
-pass
-m_file = open("data.txt")
-final_file = open("report.txt", 'w')
-n = input(int)
-text = m_file.read(n)
-freq_dict = calculate_frequencies(text)
-stop_words = input(tuple)
-sort = filter_stop_words(freq_dict, stop_words)
-N = input(int)
-output = get_top_n(sort, N)
-final_file.write(output)
-m_file.close()
-final_file.close()
+def write_to_file(path_to_file, content):
+    text = open(path_to_file, "w")
+    text.write(content)
```

Comment:
@gradevonn why didn't you show me your work today? Seems like it is working here not correct - content is supposed to be a tuple (top_n from function get_top_n) and in current implementation I get 'write() argument must be str, not tuple'. Read description carefully and notice me if your work needs to be checked.